ts/containers/connect_wallet_dialog.ts

```diff
@@ -28,6 +37,21 @@ const mapDispatchToProps = (dispatch: Dispatch<State>): ConnectedDispatch => {
         onDismiss: (): void => {
             dispatcher.updateIsConnectWalletDialogOpen(false);
         },
+        onConnectWallet: async (fallbackNetworkId: Network): Promise<void> => {
+            try {
+                dispatcher.updateIsConnectWalletDialogOpen(false);
+                const providerState = providerStateFactory.getInitialProviderState(fallbackNetworkId);
+
+                const networkId = await providerState.web3Wrapper.getNetworkIdAsync();
+                if (networkId !== fallbackNetworkId) {
+                    dispatcher.updateNetworkId(networkId);
+                }
+
+                await asyncDispatcher.fetchAccountInfoAndDispatchToStore(providerState, dispatcher, true);
+            } catch (err) {
+                // TODO(kimpers): handle errors
```

Comment:
We need a way to handle error messages. There is a SnackBar component already in the project that we could utilize. Wdyt?